src/circuit/gadgets/constraints.rs

```diff
@@ -698,6 +799,172 @@ mod tests {
 
     proptest! {
 
+        #[test]
+        fn test_enforce_equal((a, b) in any::<(FWrap<Fr>, FWrap<Fr>)>()) {
+            prop_assume!(a != b);
```

Comment:
Why make this assumption? More to the point, why engineer this test (with two random field elements) such that they (for all practical purposes) never *will* be equal? Isn't the case where they are equal an important one to test?

Reply:
Yes, if `a` is equal to `b` the negative test would pass, although it shouldn't, given its purpose. It should test the constraint system fail when `a` is not equal to `b`, so we assume this condition is achieved using `prop_assume`.

Reply:
Yeah, I think I'm inside the logic now. I should note that this the `.0` combined with the short names end up making it very hard to visually distinguish. For example, my rant below was triggered by a pure misreading on my part. Totally my fault! Nevertheless, even now knowing what's happening. I find it very hard to look at the cases below and distinguish them visually. Is there an ergonomic solution to this?

Reply:
Yes, the negative test would pass, but it shouldn't pass in this situation.

Reply:
Totally agree that in practice it would not happen, but that is the whole purpose of the test, so it seemed to be clearer if we assume `a` is different than `b`.